### PR TITLE
Check that rc != 0 for steps with 'failed_when'

### PR DESCRIPTION
**Pull request by vrutkovs**
Some parts don't check the result rc at all, some check that 'error' is present there.
Instead the safest way is to check for both

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1534538

### CODE REVIEW

File: roles/calico_master/tasks/main.yml [hunk 1]
```diff
@@ -23,7 +23,7 @@
     -f {{ mktemp.stdout }}/calico-policy-controller.yml
     --config={{ openshift.common.config_base }}/master/admin.kubeconfig
   register: calico_create_output
-  failed_when: ('already exists' not in calico_create_output.stderr) and ('created' not in calico_create_output.stdout)
+  failed_when: "('already exists' not in calico_create_output.stderr) and ('created' not in calico_create_output.stdout) and calico_create_output.rc != 0"
```
Comment (sdodson) [inline roles/calico_master/tasks/main.yml]: Do you mind changing all of these conditionals to be lists? We're trying to convert all conditionals to lists whenever we touch them.

```
failed_when:
  - a 
  - not b
  - c or d
```

File: roles/openshift_metrics/tasks/oc_apply.yaml [hunk 2]
```diff
@@ -16,7 +16,9 @@
     apply -f {{ file_name }}
     -n {{namespace}}
   register: generation_apply
-  failed_when: "'error' in generation_apply.stderr"
+  failed_when:
+    - "'error' in generation_apply.stderr"
+    - "generation_apply.rc != 0"
```
Comment (sdodson) [inline roles/openshift_metrics/tasks/oc_apply.yaml]: like this. :+1:

File: roles/calico_master/tasks/main.yml (hunk 1 above)
Comment (vrutkovs) [reply roles/calico_master/tasks/main.yml]: I don't think its possible in this case. The list would turn these `and` in `or`'s effectively.

If its just for readability I could use slashes probably

Comment (sdodson) [reply roles/calico_master/tasks/main.yml]: No, lets not do slashes. The way I'm reading that conditional it'd just be
```
failed_when:
  - 'already exists' not in calico_create_output.stderr
  - 'created' not in calico_create_output.stdout
  - calico_create_output.rc != 0
```

Anyway, we'll minimize change and accept it as is.

**Pull request merged**